File: COURSE 1 (Algorithmic Toolbox)/week4_divide_and_conquer/0.Week 4 assignment/src/com/company/MajorityElement.java
package com.company;

import java.util.*;
import java.io.*;

public class MajorityElement {

    private static int getMajorityElement(int[] a) {
        Arrays.sort(a);
        int count=1;
        for(int i=1;i<a.length;i++){
            if(a[i]==a[i-1]){
                count++;
                if(count>a.length/2){
                    return 1;
                }
            }else {
                count=1;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        FastScanner scanner = new FastScanner(System.in);
        int n=scanner.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = scanner.nextInt();
        }
        if (getMajorityElement(a) != -1) {
            System.out.println(1);
        } else {
            System.out.println(0);
        }
    }

    static class FastScanner {
        BufferedReader br;
        StringTokenizer st;

        FastScanner(InputStream stream) {
            try {
                br = new BufferedReader(new InputStreamReader(stream));
            } catch (Exception e) {
                e.printStackTrace();
            }
        }

        String next() {
            while (st == null || !st.hasMoreTokens()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }
    }

}

File: COURSE 1 (Algorithmic Toolbox)/week4_divide_and_conquer/0.Week 4 assignment/Submitted files c++/majority_element.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
using std::vector;

int get_majority_element(vector<int> &a) {
    sort(a.begin(),a.end());
    int count=1;
    for(int i=1;i<a.size();i++){
        if(a[i]==a[i-1]){
            count++;
            if(count>a.size()/2){
                return 1;
            }
        }else {
            count=1;
        }
    }
    return -1;
}

int main() {
  int n;
  std::cin >> n;
  vector<int> a(n);
  for (size_t i = 0; i < a.size(); ++i) {
    std::cin >> a[i];
  }
  if(get_majority_element(a)!=-1){
    std::cout << 1 << '\n';
  }else{
        std::cout << 0 << '\n';

  }
}
